### ISSUE TITLE
L2 Gas Bug.

### ISSUE THREAD

**Issue by r1cs** [ethereum-optimism/optimism#1319]
https://github.com/ethereum-optimism/optimism/blob/2f4c4846a6e0074f1df1690c2cd46a1f908f3781/packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol#L1650

As far as you guys siad in this function:
**a contract could pass in just enough gas to cause the INVALID_STATE_ACCESS check to fail on L1 but not on L2**.
If this happened, which means the attacker can use this feature"dismatched gas consume between L1 and L2" , to do more powerful attack.

**Comment by smartcontracts** [comment]
Yes, but this is why we enforce the following: https://github.com/ethereum-optimism/optimism/blob/2f4c4846a6e0074f1df1690c2cd46a1f908f3781/packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol#L1662

I don't think this is actually a bug. Can you provide additional detail?

**Comment by r1cs** [comment]
sure.In l2geth, the stateManager's function dont consume the gas, so the gas consumed will not dismatch between l1 and l2, this featur may  be used by attacker

**Comment by r1cs** [comment]
sure.In l2geth, the stateManager's function dont consume the gas, so the gas consumed will dismatch between l1 and l2, this featur may be used by attacker
